Huge rewrite of project

Xentraxx opened the pull request:
Heyho!

Thank you for this great project!

I did a huge rewrite of the project with a lot of added functionality and with 200 unit and functional tests.
Main improvements are the ability to write exif data (DateTime and GPS) and support for way more file types!
Please check out the details in the changelogs.

It includes many bug fixes and I am actively working on it, so everyone is welcome to give my fork a try.

TheLastGimbus commented:
My main objective for gpth is to be very stable and careful - i may look at all of this some day, but the current version seems to be doing really great so i stick with it for now :ok_hand:  - big thanks for all of this, it may help someone (maybe me, when i will put some time into it)

mian196 commented:
hey does it support supplemental-metadata.json?

Gong-Bao-Chicken commented:
I have used this version and found it to be working stable. 
Thank you for the time and effort put into this!

Xentraxx commented:
Thank you! Happy to hear that :)

Xentraxx commented:
For everyone interested and willing to test and give feedback: https://github.com/Xentraxx/GooglePhotosTakeoutHelper

jaimetur committed:
Fixed: Duplicates files are now removed even if they belong to different Albums, but the primaryFile of the entity keeps all the Albums path for all the duplicates and in Step 7 (Move files) it re-create the copy in each Album based on the selected strategy.

jaimetur committed:
Added some comments to the media_entity_collection code for clarification

jaimetur committed:
Added some comments to the media_entity_collection code for clarification

Xentraxx committed:
linting fixes

Xentraxx committed:
Merge pull request #74 from jaimetur/v4.3.0-development

V4.3.0 by Jaime Tur

Xentraxx committed:
Version fix

jaimetur committed:
Created v4.3.1 and Improve Performance in Remove Duplicates Step

jaimetur committed:
Change README.md to add Star History & Contributors History

jaimetur committed:
Fixed typo in README.md

jaimetur committed:
Fixed typo in README.md

jaimetur committed:
Fixed typo in README.md

jaimetur committed:
More aggressive Optimization in Find Duplicates Step. Now two pre-buckets phases are applied to reduce the probability of collision before to calculate full hashes.

jaimetur committed:
Merge remote-tracking branch 'origin/v4.3.1-development' into v4.3.1-development

jaimetur committed:
Created v4.4.0

jaimetur committed:
Structure Refactoring: Phase 01

jaimetur committed:
Structure Refactoring: Phase 02

jaimetur committed:
Structure Refactoring: Phase 03

jaimetur committed:
Structure Refactoring: Phase 04

jaimetur committed:
Structure Refactoring: Phase 05

jaimetur committed:
Structure Refactoring: Phase 06

jaimetur committed:
Structure Refactoring: Phase 07

jaimetur committed:
Structure Refactoring: Phase 08

jaimetur committed:
Structure Refactoring: Phase 09

jaimetur committed:
Structure Refactoring: Phase 10 (Finished)

jaimetur committed:
Code Structure refactored for a better understanding and easier way to find each module. Created a single package gpth-lib with all the exported modules for an easier way to manage imports and refactoring.

Xentraxx committed:
version bump to 4.3.1

Xentraxx committed:
added question for album behaviour to interactive mode

Addresses #76

jaimetur committed:
Merge branch 'v4.3.1-development' into v4.3.1-development

Xentraxx committed:
Merge pull request #80 from jaimetur/v4.3.1-development

Improve Performance in Remove Duplicates Step
Change README.md to add Star History & Contributors History

Xentraxx committed:
linting fixes

Xentraxx committed:
removed duplicate table from readme

jaimetur committed:
Actualizar media_entity_collection.dart

jaimetur committed:
Code Refactored to isolate the execution logic of each step into the .execute() function of the step's class. In this way the media_entity_collection module is much clearer and easy to understand and maintain.

jaimetur committed:
Code Refactored to isolate the execution logic of each step into the .execute() function of the step's class. In this way the media_entity_collection module is much clearer and easy to understand and maintain.

jaimetur committed:
Support for 7zip and unzip extractors (if found in your system). This is shy the native extractor does not extract properly filenames or dirnames with latin chars.

jaimetur committed:
Support new `Extra` files from Google Takeout with following suffixes: `-motion`, `-animation`, `-collage`

jaimetur committed:
Fix some folder names in the new folder structure

jaimetur committed:
Fix some folder names in the new folder structure

jaimetur committed:
Fix issues with Albums names which ends with space

jaimetur committed:
Fix issues with Albums names which ends with space

jaimetur committed:
Fix issues with Step 5 during Exiftool call

jaimetur committed:
Added Backward-compatibility façade in MediaEntityCollection to add legacy compatibility with old methods that now delegate to each Step's execute (used by tests)

jaimetur committed:
New flag `--keep-input` to Work on a temporary sibling copy of --input (suffix _tmp), keeping the original untouched.

jaimetur committed:
New flag `--keep-input` to Work on a temporary sibling copy of --input (suffix _tmp), keeping the original untouched.

jaimetur committed:
New flag `--keep-input` to Work on a temporary sibling copy of --input (suffix _tmp), keeping the original untouched.

jaimetur committed:
Always clean Output Directory in no-interactive mode and in interactive mode if user chose this option

jaimetur committed:
Added logging messages to explain when should clone input folder when --keep-input flag is given

jaimetur committed:
Fix found and merged albums message after Step 6

jaimetur committed:
Step 5 (Write EXIF) Optimization after refactoring done. Added more counters to the log.

jaimetur committed:
Step 5 (Write EXIF) Optimization after refactoring done. Added more counters to the log.

jaimetur committed:
Step 5 (Write EXIF) Optimization after refactoring done. Added more counters to the log.

jaimetur committed:
Added flag moveDuplicatesToDuplicatesFolder to GlobalConfigService

jaimetur committed:
Create v5.0.0 (MediaEntity changed)

jaimetur committed:
Test Fixed

jaimetur committed:
Logs homogenized for all Steps

jaimetur committed:
Updated CHANGELOG.md

jaimetur committed:
Actualizar gpth.dart

jaimetur committed:
Fix need to clone inputDir log messages

jaimetur committed:
Some lint fixing

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Created GitHub Action `build-and-create-release.yml`

jaimetur committed:
Change from logDebug to logInfo the statistics lines on Step 4 and 5 to avoid PhotoMigrator skip it when not run with logLevel=debug

jaimetur committed:
removed "v" before version from binaries

jaimetur committed:
Changed moveDuplicatesToDuplicatesFolder from GlobalConfig to RuntimeConfig and allow user to choose its value using flag `--keep-duplicates` (default disabled)

jaimetur committed:
Changed moveDuplicatesToDuplicatesFolder from GlobalConfig to RuntimeConfig and allow user to choose its value using flag `--keep-duplicates` (default disabled)

jaimetur committed:
Updated CHANGELOG

jaimetur committed:
Performance Optimization in Step 3 (Remove Duplicates)

jaimetur committed:
Performance Optimization in Step 3 (Remove Duplicates)

jaimetur committed:
Performance Optimization in Step 3 (Remove Duplicates)

jaimetur committed:
Created 5.0.2 to introduce a new code re-design.

jaimetur committed:
Code Re-Design: Phase 1

jaimetur committed:
Code Re-Design: Phase 2

jaimetur committed:
Code Re-Design: Phase 3

jaimetur committed:
Code Re-Design: Phase 3

jaimetur committed:
Code Re-Design: Phase 3

jaimetur committed:
Code Re-Design: Phase 3

jaimetur committed:
Updated CHANGELOG.md

jaimetur committed:
Fixed Linting issues and Adapted Test to new structure

jaimetur committed:
Fixed Linting issues and Adapted Test to new structure

jaimetur committed:
Fix partner_sharing_integration_test.dart

jaimetur committed:
Fix partner_sharing_integration_test.dart

jaimetur committed:
Renamed MediaEntity attribute `belongToAlbums` to `albumsMap`

jaimetur committed:
Fixed symlink_service.dart that was not working on linux platforms. Optimized Step3 (Find Duplicates). Increase max concurrency limit for `diskOptimized` operations from 8 to 32.

jaimetur committed:
Fixed file_operation_service.dart that was not working on linux platforms generating a wrong normalized path (without initial `/`.

jaimetur committed:
Fixed file_operation_service.dart that was not working on linux platforms generating a wrong normalized path (without initial `/`.

jaimetur committed:
Fixed file_operation_service.dart that was not working on linux platforms generating a wrong normalized path (without initial `/`.

jaimetur committed:
Fixed zip_extraction_service.dart to support extract UTF-8/latin1 chars on folder/files names.

jaimetur committed:
Fixed zip_extraction_service.dart to support extract UTF-8/latin1 chars on folder/files names.

jaimetur committed:
Fixed zip_extraction_service.dart to support extract UTF-8/latin1 chars on folder/files names.

jaimetur committed:
Fixed zip_extraction_service.dart to support extract UTF-8/latin1 chars on folder/files names.

jaimetur committed:
Rename Step 3 from `Remove Duplicates` to `Merge Media Entities` because this is the main purpose of this step. Added fallback to hardlink/junctions for windows if native API to `WindowsSymlinkService` for symlink creation fails (due to admin rights)

jaimetur committed:
Updated CHANGELOG.md. Improved logs statistics. Changed `enableBatches` GlobalConfig flag by `enableExifToolBatch`. Added new flag `enableTelemetryInMergeMediaEntitiesStep` in GlobalConfig to enable/disable Telemetry in Step 3: Merge Media Entities.

jaimetur committed:
Fix duplicate break line in log

jaimetur committed:
Added new Flag `--save-log` to activate log saving into disk.

jaimetur committed:
Added new Flag `--save-log` to activate log saving into disk.

jaimetur committed:
Change from logInfo to logDebug all internal traces of DuplicateDetectionService

jaimetur committed:
renamed argument `--fileDates` to `json-dates` to maintain name convention

jaimetur committed:
renamed argument `--fileDates` to `json-dates` to maintain name convention

jaimetur committed:
fix typo

jaimetur committed:
fix typo

jaimetur committed:
Step 8 (Update creation time) now is multi-platform. Performance optimization in Step 3.

jaimetur committed:
Update CHANGELOG.md and restored Header comments on each Step.

jaimetur committed:
Added printPlain() method to LoggingService class and logPrint() method to LoggerMixin class to emulate a print command (with no color), but writing the message into the log file.

jaimetur committed:
Aligned [logLevel] mark in all logging messages

jaimetur committed:
Added `FallbackCombinedTried` to Stats in Step 7.

jaimetur committed:
Added `FallbackCombinedTried` to Stats in Step 7. Fixed GPS write stats.

jaimetur committed:
Fix duplicate_detection_service_test.dart

jaimetur committed:
Fix linting issue in logging_service.dart

jaimetur committed:
Added Concurrency info to Step 3

jaimetur committed:
Restored Extraction methods stats in the Final Summary

jaimetur committed:
Prevent LoggerMixin create multiple instances

jaimetur committed:
Phase 1: Replace print messages by LoggerMixin methods

jaimetur committed:
Phase 2: Replace print messages by LoggerMixin methods in all Steps

jaimetur committed:
Phase 3: Replace print messages by LoggerMixin methods in all Steps

jaimetur committed:
Phase 4: Replace print messages by LoggerMixin methods in all Steps

jaimetur committed:
Fixing Tests: Phase 1 (Unit tests) - All OK

jaimetur committed:
Fixing Tests: Phase 2 (Integration tests) - 75% done

jaimetur committed:
Fixing Tests: Phase 3 (Integration tests) - 95% done (320/337 passed; 1 skipped; 16 failed)

jaimetur committed:
Fixing Tests: Phase 3 (Integration tests) - 95% done (320/337 passed; 1 skipped; 16 failed)

jaimetur committed:
Fixing Tests: Phase 3 [Integration tests - 95% done (320/337 passed; 1 skipped; 16 failed)]; [Unit Tests - 100% done (313/35 passed; 2 skipped; 0 fails)]; [e2e Tests - Not tested yet]; [Performance Tests - Not tested yet]

jaimetur committed:
Fixing Tests: Phase 3 [Integration tests - 95% done (320/337 passed; 1 skipped; 16 failed)]; [Unit Tests - 100% done (313/35 passed; 2 skipped; 0 fails)]; [e2e Tests - Not tested yet]; [Performance Tests - Not tested yet]

jaimetur committed:
Improve Telemetry and logs on Step 7 (Write EXIF)

jaimetur committed:
Change Log Dir to output folder instead of `Log` folder

jaimetur committed:
Change Log Dir to output folder instead of `Log` folder

jaimetur committed:
Added new attribute `isMoved`to FileEntity and added new methods `getSecondary()`, `getPrimaryAndSecondary()`, `getDuplicates()`, `getAllFiles()` to MediaEntity

jaimetur committed:
Fixed bug in WriteExif when running ExifTool in batches where there are mixing of Tags in each file of the same batch, resulting to assign wrong tags with wrong dates to some files in the batch. Added new attribute `isDeleted`to FileEntity

jaimetur committed:
Included warnings/errors summary step-by-step results and processing summary in the log.

jaimetur committed:
Added new attribute `isDuplicateCopy`to FileEntity

jaimetur committed:
Updated CHANGELOG.md

jaimetur committed:
New album moving strategy `ignore-albums` to completely ignore all Albums content. Moving Strategies re-defined. Added `reverse-shortcut` strategy to interactive mode

jaimetur committed:
New album moving strategy `ignore-albums` to completely ignore all Albums content. Moving Strategies re-defined. Added `reverse-shortcut` strategy to interactive mode

jaimetur committed:
New album moving strategy `ignore-albums` to completely ignore all Albums content. Moving Strategies re-defined. Added `reverse-shortcut` strategy to interactive mode

jaimetur committed:
Fix inconsistencies in moving_strategies.dart to align with specification.

jaimetur committed:
Fixing Tests: Phase 3 [Integration tests - 95% done (322/337 passed; 1 skipped; 14 failed)]; [Unit Tests - 100% done (313/35 passed; 2 skipped; 0 fails)]; [e2e Tests - Not tested yet]; [Performance Tests - Not tested yet]

jaimetur committed:
Updated README.md with new album strategy description

jaimetur committed:
Updated README.md with new album strategy description

jaimetur committed:
Fixed exiftool_service.dart to avoid IFD0 pointer references and use of -common_args when @ARG_FILE is used. Also added support to PNG using XMP instead of EXIF

jaimetur committed:
Fixed exiftool_service.dart to avoid IFD0 pointer references and use of -common_args when -@A RGFILE is used. Also added support to PNG using XMP instead of EXIF

jaimetur committed:
Fixed exiftool_service.dart to avoid IFD0 pointer references and use of -common_args when -@A ARGFILE is used. Also added support to PNG using XMP instead of EXIF. Included Timeouts on ExifTool operations.

jaimetur committed:
Now when the batch has a single file, call the single method so fallback metrics (Recovered/Fail) are accounted correctly in ExifWriterService. Also fallback fails are now counted if used batches when the batch only contains 1 file (at the end a batch with some file failing will end in a single-file batch).

jaimetur committed:
Changed log name from `gpth-{version}_{timestamp}.log` to `gpth_{version}_{timestamp}.log`

jaimetur committed:
Updated README.md

jaimetur committed:
Fix `Bad state: Process is detached` on Exiftool writer. Added second progress bar to see the progress of the final flush.

jaimetur committed:
Merge remote-tracking branch 'origin/5.0.4' into 5.0.4

jaimetur committed:
Fix `Bad state: Process is detached` on Exiftool writer. Added second progress bar to see the progress of the final flush.

jaimetur committed:
Linting fixes

jaimetur committed:
Enabled log saving by default

jaimetur committed:
Renamed strategy `ignore-albums` by just `ignore`. Fixed error `No MOVE operation emitted by strategy for primary file` while using `ignore` strategy, since this strategy does not emit MOVE operation for all primary file, it emits DELETE operation for non-canonical primaries (files within album only)

jaimetur committed:
Forced log path to use output dir. Fixed file_extension_corrector_service.dart to detect paths ending with spaces

jaimetur committed:
Added progress bar to Step 3 (Merge Media Entities)

jaimetur committed:
Added progress bar to Step 3 (Merge Media Entities)

jaimetur committed:
A (ExifToolService): I added -F to the common arguments (_commonWriteArgs). It’s an immediate patch that often turns “Truncated InteropIFD” into a success.

B (Step 7): If we detect a “problematic” JPEG, we force XMP (CreateDate/DateTimeOriginal/ModifyDate + signed GPS), both when initially building the tags (via _forceJpegXmp) and again on retry when a batch fails and stderr contains Truncated InteropIFD (in-place conversion of those entries with _retagEntryToXmpIfJpeg).

C (Step 7): Smart split in writeBatchSafe: we parse stderr, separate only the conflicting files, retry the rest in a single batch, and write the conflicting ones per-file (without blocking progress). If paths can’t be extracted, we fall back to your original recursive split.

jaimetur committed:
1. (ExifToolService): I added -F to the common arguments (_commonWriteArgs). It’s an immediate patch that often turns “Truncated InteropIFD” into a success.

2. (Step 7): If we detect a “problematic” JPEG, we force XMP (CreateDate/DateTimeOriginal/ModifyDate + signed GPS), both when initially building the tags (via _forceJpegXmp) and again on retry when a batch fails and stderr contains Truncated InteropIFD (in-place conversion of those entries with _retagEntryToXmpIfJpeg).

3. (Step 7): Smart split in writeBatchSafe: we parse stderr, separate only the conflicting files, retry the rest in a single batch, and write the conflicting ones per-file (without blocking progress). If paths can’t be extracted, we fall back to your original recursive split.

4. Changed default value for flag `--update-creation-time. Now is enabled by default.

jaimetur committed:
Zip extraction service now uses
1. Unzip if available (only linux/mac)
2. 7-Zip as fallback if available
3. Native extractor

jaimetur committed:
Added Progress bar to Step 1 & Step 2

jaimetur committed:
New release v5.0.5

jaimetur committed:
Separated Step's logic in the associated service to keep the step module as a wrapper only and centralize all the services used for each step.ac

jaimetur committed:
Fixed _extractBadPathsFromExifError method to detect from exiftool output bad files with relative paths

jaimetur committed:
Added Total time to Telemetry Summary in Step 3

jaimetur committed:
Changed Format of Telemetry in Step 4. Added percentage to all progress bars.

jaimetur committed:
Fixed duplicated symlinks in Albums when a file belong to more than 1 album

jaimetur committed:
Linting fixing

jaimetur committed:
Fixed duplicated files in Albums when a file belong to more than 1 album (affected strategies: shortcut, reverse-shortcut & duplicate-copy)

jaimetur committed:
- Added support for Special Folders management such as `Archive`, `Trash`, `Locked folder`. Now those folders are excluded from all album strategies and are moved directly to the output folder.

jaimetur committed:
- Minor methods refactoring in ExifToolService

jaimetur committed:
Minor typo

jaimetur committed:
Minor typo

jaimetur committed:
Minor typo

jaimetur committed:
Added saveLog (read from flags) to ConfigBuilder

jaimetur committed:
Fixed error decoding Exiftool output with UTF-8/latin chars

jaimetur committed:
Performance Improvements in `step_03_merge_media_entities_service.dart`. Now grouping method can be easily changed. Internal `_fullHashGroup` is now used instead of 'groupIdenticalFast' to avoid calculate buckets again.

jaimetur committed:
Performance Improvements in `step_03_merge_media_entities_service.dart`. Now grouping method can be easily changed. Internal `_fullHashGroup` is now used instead of 'groupIdenticalFast' to avoid calculate buckets again.

jaimetur committed:
Fix exiftool reader fails on path with mojibake.

jaimetur committed:
Updated CHANGELOG.md

jaimetur committed:
Updated CHANGELOG.md

jaimetur committed:
Minor change in Step 4 Telemetry stats

jaimetur committed:
Minor change in Step 4 Telemetry stats

peter9811 commented:
+1

jaimetur committed:
Performance improvement in Step3.

jaimetur committed:
Minor code refactoring on step_03_merge_media_entities_service.dart

jaimetur committed:
Step 3 Summary modified.

Xentraxx committed:
removed 7zip folder

Xentraxx committed:
automatic linting fixes

Xentraxx committed:
Merge branch 'pr/83' into prerelease

Xentraxx committed:
linting fixes and formatting

Xentraxx committed:
linting fixes

Xentraxx committed:
readme update for 5.0.5 prerelease

Xentraxx committed:
Fix typo in README regarding v5.0.5 features

jaimetur committed:
Auto-Resume Support: Phase 1

jaimetur committed:
Auto-Resume Support: Phase 2

jaimetur committed:
Auto-Resume Support: Phase 3

jaimetur committed:
Merge branch 'master' into 5.0.6

jaimetur committed:
Merge pull request #8 from jaimetur/5.0.6

5.0.6 - Added Auto-Resume Support

jaimetur committed:
Auto-Resume Support: Phase 4

jaimetur committed:
Merge remote-tracking branch 'origin/master' into 5.0.6

jaimetur committed:
Auto-Resume Support: Phase 4

jaimetur committed:
Merge branch 'prerelease' into 5.0.6

jaimetur committed:
Linting fixing

jaimetur committed:
- New feature to group Untitled Albums into `Untitled Albums` folder.

Xentraxx committed:
dart format

Xentraxx committed:
linting fixes

Xentraxx committed:
removed old intel macos runner

Xentraxx committed:
Update .gitignore

Xentraxx committed:
Merge pull request #89 from jaimetur/5.0.6

prerelease of v5.0.6

Xentraxx committed:
upgraded to exif_reader 4.0.2 package

Xentraxx committed:
fixed #90

Xentraxx committed:
fixed that native exif writer not used when exiftool not installed

Fixed major error which led to native exif write methods not being used when exiftool was not installed. See #95

Xentraxx committed:
temporarily changed the image library to my fixed fork

Should fix #95 until my pull request is accepted in the source repo

Xentraxx committed:
Update CHANGELOG.md

Xentraxx committed:
small changes in readme and changelog

Xentraxx committed:
Merge branch 'v.5.0.6-development'

Xentraxx committed:
dart format

Xentraxx committed:
added internal config flag to disable resume config for tests

Also fixed some tests and formatted everything

Xentraxx committed:
test fixes

Xentraxx committed:
updated bug report issue template

Xentraxx committed:
added feature github issue template

Xentraxx committed:
fixed german unknown folder name

Xentraxx committed:
Update CHANGELOG.md

Xentraxx committed:
version number update

Xentraxx committed:
fixed github workflow artifact naming to avoid double version in name

Xentraxx committed:
updated readme with reference to wiki

Xentraxx committed:
added check for disableResumeCheck to Step 7

Xentraxx committed:
fixed typo

Xentraxx committed:
fixed partner sharing?

Xentraxx committed:
dart format

Xentraxx committed:
Update CHANGELOG.md

Xentraxx committed:
updated image to patched version

Xentraxx committed:
implemented feature request #88

Xentraxx committed:
Prevent auto-clean of output dir in non-interactive mode

Refuse to automatically clean the output directory in non-interactive mode to avoid accidental deletion of unrelated user files. The program now exits with a clear message instructing users to manually clean or select an empty directory.
Related but not the source of #98

Xentraxx committed:
Add safety checks and warnings for directory cleanup and extraction

Enhanced output directory cleanup with explicit warnings in interactive mode to inform users about potential data deletion. Updated zip extraction service to refuse extraction into non-empty directories for safety, preventing accidental data loss. Improved user messaging to emphasize the requirement for an empty extraction folder. #98

Xentraxx committed:
Update changelog for version 5.0.7

Added notes for bug fixes and new features in version 5.0.7, including safer ZIP extraction behavior, improved interactive mode warnings, and restoration of truncated media filenames from JSON sidecars.

Xentraxx committed:
Log invocation details in LoggingService

Adds functionality to capture and log the executable, arguments, and working directory of the process invocation in the LoggingService. Updates the entrypoint to set these details early, and enhances log headers with invocation context for improved traceability. Also makes the 'flag combination' field optional in the bug report template.

Xentraxx committed:
updated image to version with major fixes

Xentraxx committed:
changelog and version bump

Xentraxx committed:
Fix UTC conversion bug in EXIF writing

Corrects handling of UTC timestamps when writing EXIF data, ensuring proper offset tags (+00:00) are set for UTC times. Updates version to 5.0.9 and adds integration test to verify UTC DateTimeOriginal and offset behavior.

Xentraxx committed:
downgraded image library temporarily again

Xentraxx committed:
Fix EXIF GPS writing and partner shared folder structure

Corrects EXIF GPS tag writing to preserve geodata using proper EXIF types, addressing a bug where geodata was removed. Updates partner shared file organization to mirror the normal folder structure under PARTNER_SHARED/ALL_PHOTOS. Enhances and expands end-to-end and integration tests to validate EXIF GPS preservation and correct output structure. Updates image dependency to 4.7.2.

Xentraxx committed:
Use platform-agnostic path joins in file operations

Replaced hardcoded Unix path separators with platform-agnostic path joins using the 'path' package in moving strategies and media entity service. Updated related tests for formatting consistency. This fixes a bug where file operations could fail on non-Unix platforms due to incorrect path separators. #96

Xentraxx committed:
Expand year folder regex to support more languages

Updated the isYearFolder method to recognize year folders with prefixes in English, Spanish, and German ("Photos from", "Fotos del", "Fotos von"). This improves internationalization support for folder classification.

Xentraxx committed:
Normalize paths to NFC and add folder patterns

Normalize stored file paths to NFC using unorm_dart to fix macOS unicode normalization issues (fixes #99). Add multilingual "Photos from" folder regex patterns and expose reusable patterns from TakeoutFolderClassifierService; update PathResolverService and interactive checks to use these patterns. Bump app version to 5.1.0 (CHANGELOG and pubspec) and update exif_reader dependency. Add comprehensive unit tests for Unicode normalization and adjust an integration test's invalid-year comment.

TheLastGimbus commented:
Hi there @Xentraxx 

I see there's a lot of work you're doing here - I'm very glad of it !

I kinda moved a lot in my life and spend much less of my time on programming - i doubt i will update og gpth soon, but if i ever would, I would probably do it myself (blazing fast in Rust or smth :zap: ) - because time spent at this was always 90% testing and 10% coding, so collaborating with someone else would take much more for me (rather than just writing it myself from the ground)

Anyway, i'm very glad you're doing all of this, but if I may, i kindly ask you to change the name of your stuff - it can include original, like... gpth-neo, or gpth-x (like vi and vim, then neovim etc...) - whatever, just make it distinguishible between original version. I'll gladly mark existance of gpth-neo somewhere in my og readme :)

Last time i checked, my good-ol-gpth-v3 worked safe and didn't mess up anyone's photos, and that's the only thing i can answer for. When someone comes to me and says "Hey! I downloaded gpth-v5 and it broke! What the hell TLG??" - i can say nothing :shrug:

---

Last thing - i still receive some donations from people using gpth - well, it got pretty famous with some articles about it, etc - if you feel like it, you can put your own donation link on your new "gpth-neo", or remove mine, or leave both of us :)

Note: once in a month, i receive a weird random mail from someone using super-old python gpth-v2, and asking why doesn't it work - that's why i want to make it clear for people what they are using and that gpth != gpth-neo

I may come back some day, if they kick me out of my collage or i get bored of my new hobbies - Keep up the good work, and cheers :tada: -

Xentraxx commented:
Hey @TheLastGimbus 
Sure I get that. It's difficult to support a project of someone else.
I changed the name to GPTH Neo.
And yes I feel you with the time needed for testing.
I therefore always just fix bugs in a way I think they should be fixed, write a test and if the test is green, I ask the person who created the issue to rerun the new version before I actually release it.
That way I save a lot of time. But also my fork should be a lot faster than the og gpth. We put a lot of effort into optimization and performance.

I wanted to leave the donation link standing until I have your permission to remove it. I removed it from my project. I personally don't want any donations.

It would be cool if you could include a shoutout to my fork in the readme of the og gpth.
I know for a fact there are a few edge cases and problems with the og gpth (e.g. outdated dependencies, see https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/issues/453), so maybe you could just state that you are not so active and if anyone experiences problems they can try my fork. That means less issues in your repo you have to deal with or which stay unanswered and more traffic to mine which helps me to get more feedback and make it more stable.

Cheers!
